web-api/src/storage/postgres.rs

```diff
@@ -1105,4 +1295,41 @@ impl storage::Tag for Storage {
             )
             .collect())
     }
+
+    async fn put(
+        &self,
+        document_id: &DocumentId,
+        tags: &[DocumentTag],
+    ) -> Result<Option<()>, Error> {
+        let mut tx = self.postgres.pool.begin().await?;
+
+        let inserted = sqlx::query_as::<_, (bool,)>(
+            "UPDATE document
+            SET tags = $1
+            WHERE document_id = (
+                SELECT document_id
+                FROM document
+                WHERE document_id = $2
+                FOR UPDATE
```

Comment:
:wrench:  This doesn't really do anything AFIK.

`FOR UPDATE` only acquires a "update" lock for the row for the duration of the transaction.

`UPDATE` does the same! (at least with how the default transaction level works in prostgres).

Furthermore update row locks only synchronize with themself, i.e. don't lock non-for-update select reads in other transactions.

Reply:
this is likely a remnant from the document migration prs. here i copied pasted the code from one of the other setters. i guess it makes sense to revisit this issue in a separate pr and address all occurences there.

Reply:
okay :pushpin:

Reply:
i added [ET-4339] to the backlog for this issue.